### ISSUE TITLE
Add main libraries documentation

### ISSUE THREAD

**Issue by leonardofalk** [Tibia-WindBot/WindBot-Docs#23]
It'd be really nice if we could add the documentation for the main libraries as we're working on the bot documentation almost a year and we haven't worked on it.

They can be found [here](https://github.com/LeonardoFalk/Libraries-Documentation).

**Comment by UltraMC** [comment]
@rmobis mentioned link is broken, did you mean https://github.com/Tibia-WindBot/Libraries-Documentation ?

**Comment by leonardofalk** [comment]
@UltraMC yeah, https://github.com/Tibia-WindBot/Libraries-Documentation is the forked version as it was back then in the first post. However some of the functions have been updated/deprecated and some of the docs were updated in each library, separated by author.

Raphael's library docs are inline and I think it has the most updated docs.

My library is being updated and the most updated version can be found in the _docfile.yml for each function, in the branch 'v2.0.0', [check this example](https://github.com/Tibia-WindBot/leonardo-library/blob/v2.0.0/src/bot/formatnumber/_docfile.yml), and it has an example file (that may be empty).

Lucas' library docs are inline too, but I think some of them may be deprecated by now, as it was recently updated.

**Comment by UltraMC** [comment]
@leonardofalk Please confirm that datasource is:
1. https://github.com/Tibia-WindBot/Libraries-Documentation
2. + some parts of it are going to be overwrited/created by adding:
2.1 https://github.com/Tibia-WindBot/raphael-library
2.2 https://github.com/Tibia-WindBot/leonardo-library
2.3 https://github.com/Tibia-WindBot/lucas-library
+ some parts in 1. and 2. are out of date anyway - with that I don't know can I do it myself

What happens in future when you add something to libraries? Where exacly it's going to be putted?

Shall I do it?

**Comment by leonardofalk** [comment]
@UltraMC the datasource is option 2, libraries.

For now it'll be put inside libraries-documentation repo.

If you wish to do so, yes, you can do it.

**Comment by UltraMC** [comment]
There's missing data - "Tibia 11" documentation is nowhere to found yet on official website we can see it.

**Comment by UltraMC** [comment]
How we want to organise it? [Preview ](http://imgur.com/a/bfUWH) for option 1

**Comment by rmobis** [comment]
@UltraMC I guess that works, thought we're on the process (kinda) of merging the three libraries into one so I'd wait on that before investing too much time in their documentation.

PS: Sorry for the delay, last few days have been rough.

**Comment by UltraMC** [comment]
@rmobis Can you describe what you have to do in "process (kinda) of merging the three libraries"?

I don't linke organasing it by categories per-creator. I would need more experienced users to categorise each function with "Event/Function/HUD/variable etc." fashion and add a new flag saying is it from core or from lib - or no flag at all if users wont care about it but IMO it's improtand as with libs I have access to their LUA source while core functions are hardcoded into software.

**Comment by rmobis** [comment]
The process of merging the three libraries is going to be a long one, but basically it means we'll stop distributing 3 library files and distribute a single on instead. This should ease contribution and maintenance from other users and also make things easier to follow.

I also agree it would be better not to separate them based on where they come from, but just throw them together in the mix with everything else and add the flags. I've had this idea in my mind for a while now...

How about we start integrating the docs from my library first (mostly because the documentation is more up to date and likely easier to parse, but also because it is the last library ran which means the functions I overwrite are the ones what really count) and see what we can do?

**Comment by UltraMC** [comment]
@rmobis Can you prepare file with map of all functions to document? After that I will work on tool/manually will handle that to exportable format.
```
file: path
name: function/variable name
category: event, variable, HUD etc.
```

Got Discord group?

**Comment by rmobis** [comment]
https://discord.gg/JTtRF

**Comment by UltraMC** [comment]
First @rmobis lib is going to be included in https://github.com/Tibia-WindBot/WindBot-Docs/tree/add-external-libs
